File: src/test/java/org/hisp/dhis/rules/models/RuleActionHideSectionTests.java
package org.hisp.dhis.rules.models;

import nl.jqno.equalsverifier.EqualsVerifier;
import nl.jqno.equalsverifier.Warning;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.JUnit4;

@RunWith( JUnit4.class )
public class RuleActionHideSectionTests
{

        @Test(expected = NullPointerException.class )
        public void createMustThrowOnNullField()
        {
                RuleActionHideSection.create( null );
        }

        @Test
        public void equalsAndHashCodeFunctionsMustConformContract()
        {
                EqualsVerifier.forClass( RuleActionHideSection.create( "test_field" ).getClass() )
                    .suppress( Warning.NULL_FIELDS )
                    .verify();
        }
}

File: src/test/java/org/hisp/dhis/rules/models/RuleActionHideSectionTests.kt
package org.hisp.dhis.rules.models

import nl.jqno.equalsverifier.EqualsVerifier
import nl.jqno.equalsverifier.Warning
import org.junit.Test
import org.junit.runner.RunWith
import org.junit.runners.JUnit4
import kotlin.test.assertFailsWith

@RunWith(JUnit4::class)
class RuleActionHideSectionTests {

    @Test
    fun createMustThrowOnNullField() {
        assertFailsWith<NullPointerException> {
            RuleActionHideSection.create(null!!)
        }
    }

    @Test
    fun equalsAndHashCodeFunctionsMustConformContract() {
        EqualsVerifier.forClass(RuleActionHideSection.create("test_field").javaClass)
                .suppress(Warning.NULL_FIELDS)
                .verify()
    }
}
